indexer-events/migrations/20230126104610_add_partition_balance.sql

```diff
@@ -0,0 +1,78 @@
+-- DONE!!! At this point the table is usable again :)
+
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202301', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2023-01-01'), fn_timestamp2nanosec(TIMESTAMP '2023-02-01'));
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202212', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2022-12-01'), fn_timestamp2nanosec(TIMESTAMP '2023-01-01'));
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202211', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2022-11-01'), fn_timestamp2nanosec(TIMESTAMP '2022-12-01'));
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202210', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2022-10-01'), fn_timestamp2nanosec(TIMESTAMP '2022-11-01'));
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202209', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2022-09-01'), fn_timestamp2nanosec(TIMESTAMP '2022-10-01'));
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202208', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2022-08-01'), fn_timestamp2nanosec(TIMESTAMP '2022-09-01'));
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202207', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2022-07-01'), fn_timestamp2nanosec(TIMESTAMP '2022-08-01'));
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202201_to_06', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2022-01-01'), fn_timestamp2nanosec(TIMESTAMP '2022-07-01'));
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202107_to_12', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2021-07-01'), fn_timestamp2nanosec(TIMESTAMP '2022-01-01'));
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202101_to_06', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2021-01-01'), fn_timestamp2nanosec(TIMESTAMP '2021-07-01'));
+SELECT fn_partition_by_range('near_balance_events', 'near_balance_events_old', 'near_balance_events_p202007_to_12', 'event_index', fn_timestamp2nanosec(TIMESTAMP '2020-07-01'), fn_timestamp2nanosec(TIMESTAMP '2021-01-01'));
+
+-- After all the data moved into the correspondent partition we can now detach and drop the old partition without the need to execute VACUUM
+-- NOTE: We could use VACUUM FULL to recover OS disk space but it would require exclusive lock
+ALTER TABLE near_balance_events DETACH PARTITION near_balance_events_old;
+DROP TABLE near_balance_events_old;
+
+-- To automatically create new partitions. 
+--  Every Monday at 11am creates a new partition for the next week
```

Comment:
Creating the partitions should be cheap. Do we want to use the same time? (I'm not sure in this, just want to know the reasoning behind this)

Reply:
No particular reason, but it's a start for the perf tests, we can change the range from week to day if needed

Reply:
I'm not about partition ranges, I'm about cron
` Every Monday at 11am`
For `fungible_token_events`, it's 10 AM

Reply:
Got it, just trying to avoid concurrent jobs, I think the default number of pg_cron jobs is 5